### PR TITLE
feat(signup): manage incomplete nic at login

### PR DESCRIPTION
**Pull request by Jisay**

### CODE REVIEW

File: src/provider.js [hunk 1]
```diff
@@ -288,7 +319,13 @@ export default function () {
         headers,
       }).done((data) => {
         self.user = data; // store user infos
-        isLogged = true;
+
+        if (data.state === 'incomplete' && !allowIncompleteNic) {
```
Comment (FredericEspiau) [inline src/provider.js]:
```suggestion
        if (data.state === SIGN_UP_STATES.incomplete && !allowIncompleteNic) {
```
?

File: src/provider.js [hunk 2]
```diff
@@ -391,6 +428,38 @@ export default function () {
...
+      const self = this;
+
+      if (!deferredObj.signUpPage) {
+        deferredObj.signUpPage = $q.defer();
+
+        // redirect to login page
+        $timeout(() => {
+          const params = [];
+
+          if (signUpUrl.indexOf('onsuccess') === -1) {
```
Comment (FredericEspiau) [inline src/provider.js]:
```suggestion
          if (!includes(signUpUrl, 'onsuccess')) { // this is Lodash
```

File: src/provider.js [hunk 3]
```diff
@@ -391,6 +428,38 @@ export default function () {
...
+            params.push(`onsuccess=${encodeURIComponent(url || $location.absUrl())}`);
+          }
+
+          if (signUpUrl.indexOf('lang') === -1) {
```
Comment (FredericEspiau) [inline src/provider.js]:
```suggestion
          if (!includes(signUpUrl, 'lang')) {
```

File: src/provider.js [hunk 4]
```diff
@@ -391,6 +428,38 @@ export default function () {
...
+            params.push(`lang=${self.user.language ? self.user.language.split('_')[0] : 'fr'}`);
```
Comment (FredericEspiau) [inline src/provider.js]:
```suggestion
            params.push(`lang=${self.user.language ? head(self.user.language.split('_')) : 'fr'}`);
```

File: src/provider.js [hunk 5]
```diff
@@ -391,6 +428,38 @@ export default function () {
...
+          }
+
+          $window.location.assign(signUpUrl + (signUpUrl.indexOf('?') > -1 ? '&' : '?') + params.join('&'));
```
Comment (FredericEspiau) [inline src/provider.js]:
```suggestion
          $window.location.assign(signUpUrl + (includes(signUpUrl, '?') ? '&' : '?') + params.join('&'));
```

File: src/provider.js [hunk 6]
```diff
@@ -391,6 +428,38 @@ export default function () {
...
+        }, 0);
```
Comment (antleblanc) [inline src/provider.js]: Not necessary to precise the `delay` (default value is `0`).

**Pull request merged**